data_scientist_skills/data_engineering.py

```diff
@@ -76,3 +80,106 @@ def process_salary_estimate(dataframe, only_mean = True):
     temp_df.drop(columns = 'salary_estimate', inplace = True)
 
     return temp_df
+
+
+###Sector Encoding Section
+
+def make_sector_group_column(df, give_group_profiles = False):
+    """Returns a df with an appended column that has the sector group number.
+    A list of dictionaries with profiles of each group can be gotten by setting
+    give_group_profiles to True.
+    """
+    ## Groups were first made using a standard method of grouping sectors found
+    ## here: https://www.investopedia.com/terms/s/sector.asp
+    ## However this led to unbalanced group sizes so I rearranged them in a way
+    ## hopefully sensible -JP
+    ## Currently they are distributed as follows:
+    ## [308, 713, 671, 674, 1694, 1203]
+
+    #Can benefit from OOP, groups can be a class object for easy modification.
+    # Further, it would help to untangle the features made ready for model to
+    # make results interpretable for the end user
+
+    temp_df = df.copy()
+    groups = [
+    ['Mining & Metals', 'Agriculture & Forestry', 'Construction, Repair & Maintenance',
+         'Oil, Gas, Energy & Utilities', 'Manufacturing','Aerospace & Defense','Transportation & Logistics',],
+
+    ['Accounting & Legal', 'Insurance','Finance', ],
+
+    [ 'Telecommunications', 'Media','Real Estate','Travel & Tourism',
+     'Arts, Entertainment & Recreation','Restaurants, Bars & Food Services','Retail',
+     'Non-Profit','Government','Education','Consumer Services',],
+
+    ['Health Care','Biotech & Pharmaceuticals',],
+
+    ['Information Technology',],
+
+    ['Business Services',]
```

Comment:
why do we need the comma at the end of a list?

Reply:
No need, it's just a result of trying different grouping combinations.

I left them for ease in trying different groupings.